api/src/Repository/Model/BaseModel.php

```diff
@@ -0,0 +1,277 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: redjik
+ * Date: 13.07.15
+ * Time: 19:24
+ */
+
+namespace Spira\Repository\Model;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\HasOne;
+use Illuminate\Database\Eloquent\Relations\HasOneOrMany;
+use Illuminate\Database\Eloquent\Relations\Relation;
+use LogicException;
+use Spira\Repository\Collection\Collection;
+
+/**
+ * Class BaseModel
+ * @package Spira\Repository\Model
+ *
+ * @method count
+ * @method get
+ * @method findOrFail
+ * @method findMany
+ *
+ */
+class BaseModel extends Model
+{
+    /**
+     * @var Relation[]
+     */
+    protected static $relationsCache = [];
+
+    /**
+     * @var BaseModel[]
+     */
+    protected $deleteStack = [];
+
+    protected $isDeleted = false;
+
+    /**
+     * @param string $key
+     * @param mixed $value
+     * @throws SetRelationException
+     */
+    public function __set($key, $value)
+    {
+        if (method_exists($this, $key)) {
+            try {
+                $value = $this->prepareValue($value);
+            } catch (\InvalidArgumentException $e) {
+                $value = false;
```

Comment:
fixed that, I was changing that place a lot so just missed that it became illogical to use exceptions
thanks for noticing